src/prefect/orion/schemas/core.py

```diff
@@ -467,6 +467,16 @@ class TaskRun(ORMBaseModel):
     def set_name(cls, name):
         return name or generate_slug(2)
 
+    @validator("cache_key")
+    def validate_cache_key_length(cls, cache_key):
+        if cache_key:
+            if len(cache_key) > PREFECT_ORION_TASK_CACHE_KEY_MAX_LENGTH.value():
+                raise ValueError(
+                    "Invalid cache_key length. The task cache key length can be modified using the PREFECT_ORION_TASK_CACHE_KEY_MAX_LENGTH setting, "
+                    "which means that you can change the maximum length of the cache_key by modifying the value of this setting on the server."
```

Comment:
I think this is a little too verbose. I'd say

"Cache key exceeded maximum allowed length of {PREFECT_ORION_TASK_CACHE_KEY_MAX_LENGTH} characters."